src/language-server/config/loadTsConfig.ts

```diff
@@ -30,36 +39,39 @@ export const loadTs: Loader = async function loadTs(filepath, content) {
 async function load(
   filepath: string,
   content: string,
-  extension: string,
+  type: "module" | "commonjs",
   compilerOptions: Partial<import("typescript").CompilerOptions>,
 ) {
-  const compiledFilepath = `${filepath}${extension}`;
   let transpiledContent;
+
   try {
-    try {
-      const config = resolveTsConfig(dirname(filepath)) ?? {};
-      config.compilerOptions = {
-        ...config.compilerOptions,
+    const config = resolveTsConfig(dirname(filepath)) ?? {};
+    config.compilerOptions = {
+      ...config.compilerOptions,
 
-        moduleResolution: typescript.ModuleResolutionKind.Bundler,
-        target: typescript.ScriptTarget.ES2022,
-        noEmit: false,
-        ...compilerOptions,
-      };
-      transpiledContent = typescript.transpileModule(
-        content,
-        config,
-      ).outputText;
-      await writeFile(compiledFilepath, transpiledContent);
-    } catch (error: any) {
-      error.message = `TypeScript Error in ${filepath}:\n${error.message}`;
-      throw error;
-    }
-    // eslint-disable-next-line @typescript-eslint/return-await
-    return await defaultLoaders[".js"](compiledFilepath, transpiledContent);
-  } finally {
-    await rm(compiledFilepath, { force: true });
+      moduleResolution: typescript.ModuleResolutionKind.Bundler,
+      target: typescript.ScriptTarget.ES2022,
+      noEmit: false,
+      ...compilerOptions,
+    };
+    transpiledContent = typescript.transpileModule(content, config).outputText;
+  } catch (error: any) {
+    error.message = `TypeScript Error in ${filepath}:\n${error.message}`;
+    throw error;
   }
+  // eslint-disable-next-line @typescript-eslint/return-await
+  const imported = await import(
+    filepath,
+    //@ts-ignore
+    {
+      with: {
```

Comment:
I’m trying to fix #225 and found out the issue is not only related to ⁠`importAttributes`. The ⁠`with` keyword is also newly introduced in new node versions.

See https://nodejs.org/api/esm.html#:~:text=This%20feature%20was%20previously%20named%20%22Import%20assertions%22%2C%20and%20using%20the%20assert%20keyword%20instead%20of%20with.%20Any%20uses%20in%20code%20of%20the%20prior%20assert%20keyword%20should%20be%20updated%20to%20use%20with%20instead.

Reply:
As these are object keys, what happens if you pass this in as an object with both the `with` and `assert` property?

Reply:
It seem fine to pass both `with` and `assert` to the import function. But I found we won't be able to access the `format` key from `importAssertions`. `importAssertions` only includes two fields despite we setting other keys on the assert object.
![CleanShot 2024-10-13 at 15 12 01@2x](https://github.com/user-attachments/assets/bfb2a857-7761-4127-8792-81e48f999171)

Reply:
Hmmm :/ 
Maybe we need to switch to `as: 'cachebust:module'` and `as: 'cachebust:commonjs'` then - or does it maybe work with a different key name than `format`? Maybe that is in some way reserved?